File: zy10420/src/main/java/com/statuspage/model/ExceptionLog.java
package com.statuspage.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "exception_logs")
public class ExceptionLog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String incidentNumber;

    @Column(nullable = false)
    private String operation;

    @Column(nullable = false)
    private String errorCode;

    @Column(length = 1000)
    private String errorMessage;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String originalInput;

    @Column(columnDefinition = "TEXT")
    private String processingConclusion;

    private String requestedBy;

    @Column(nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(nullable = false)
    private Boolean resolved = false;

    private LocalDateTime resolvedAt;

    private String resolvedBy;
}

File: xy10905/src/entities/ExceptionLog.ts
import { Entity, PrimaryGeneratedColumn, Column, CreateDateColumn } from "typeorm";

@Entity()
export class ExceptionLog {
  @PrimaryGeneratedColumn("uuid")
  id: string;

  @Column()
  requestId: string;

  @Column({ type: "text" })
  requestPath: string;

  @Column({ type: "text" })
  requestMethod: string;

  @Column({ type: "text" })
  rawInput: string;

  @Column({ type: "text" })
  errorMessage: string;

  @Column({ type: "text", nullable: true })
  errorStack: string;

  @Column({ type: "text", nullable: true })
  handlingConclusion: string;

  @Column({ default: false })
  isResolved: boolean;

  @Column({ type: "text", nullable: true })
  resolvedBy: string;

  @Column({ type: "datetime", nullable: true })
  resolvedAt: Date;

  @CreateDateColumn()
  createdAt: Date;
}
